### PR TITLE
refactor(core): Extract ExecuteContext out of NodeExecutionFunctions (no-changelog)

### PR DESCRIPTION
**Pull request by netroy**
## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/core/src/node-execution-context/execute-context.ts [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+	getRequestHelperFunctions,
+	getBinaryHelperFunctions,
+	getSSHTunnelFunctions,
+	getFileSystemHelperFunctions,
+	getCheckProcessedHelperFunctions,
+} from '@/NodeExecuteFunctions';
+
+import { BaseExecuteContext } from './base-execute-context';
+
+export class ExecuteContext extends BaseExecuteContext implements IExecuteFunctions {
```
Comment (tomi) [inline packages/core/src/node-execution-context/execute-context.ts:42]: This is somewhat confusing. We have `ExecuteContext` that extends `BaseExecuteContext` that extends `NodeExecutionContext`. Could we have some more meaningful names for these? I don't have enough context to understand the difference between them

File: packages/core/src/node-execution-context/node-execution-context.ts [hunk 2]
```diff
@@ -101,6 +130,255 @@ export abstract class NodeExecutionContext implements Omit<FunctionsBase, 'getCr
...
+				runIndex,
+				workflow,
+			} as ICredentialsExpressionResolveValues;
+		}
+
+		const nodeCredentials = node.credentials
+			? node.credentials[type]
+			: ({} as INodeCredentialsDetails);
+
+		// TODO: solve using credentials via expression
```
Comment (tomi) [inline packages/core/src/node-execution-context/node-execution-context.ts:232]: Is this created in this PR or copy-pasta?

Comment (netroy) [reply packages/core/src/node-execution-context/node-execution-context.ts:232]: copy-paste

File: packages/core/src/node-execution-context/execute-context.ts (hunk 1 above)
Comment (netroy) [reply packages/core/src/node-execution-context/execute-context.ts:42]: I want to rename all these, and add extensive docs on each of these classes, and all of the public methods. But I want to do that in https://linear.app/n8n/issue/CAT-280/add-inline-docs-for-all-the-context-objects

**Review by tomi — APPROVED** [review]
🚀

**Comment by janober** [comment]
Got released with [`n8n@1.70.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.70.0)

**Pull request merged**